hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/DualNetworkHapiTest.java

```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: Apache-2.0
+package com.hedera.services.bdd.junit;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/** Marks a test factory that should execute with two isolated subprocess networks. */
+@Target(ElementType.METHOD)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface DualNetworkHapiTest {
+    String primaryName() default "PRIMARY";
+
+    String peerName() default "PEER";
+
+    int primarySize() default 4;
+
+    int peerSize() default 4;
+
+    long shard() default 0L;
+
+    long realm() default 0L;
```

Comment:
Would prefer not to let the shard/realm be specified this way. A lot of work has been done to hide shard/realm from the test framework except where absolutely necessary; having these properties would allow the abstraction to leak even more. 

Additionally, the default network behavior is to launch before this annotation is processed. Of course dual network tests aren't default network behavior, but the default network will launch anyway, and then when this annotation is processed the test framework has to "switch gears" to redefine the networks being used.